### ISSUE TITLE
LAN blocked when using VPN - Feature?

### ISSUE THREAD

**Issue by ghost** [ukanth/afwall#768]
For the life of me i cannot find a way to allow LAN traffic when connected via WiFi while connected to VPN.   If this is something easy to allow within the configurations then please tell me how?
Ive tried allowing LAN access to a specific app but it cannot access the LAN.

How to allow access to LAN globally while VPN is connected??

if there is no configuration for this. Is it possible you can add an option to enable global LAN throughput while VPN is connected?

**Comment by facled** [comment]
i have here the same problem.
vpn is activated and afwall blocks my lan connection because it want send over vpn
wlan address is 192.168.0.10 and sends to 192.168.0.2 but afwall logs shows me: destination is 192.168.0.2 and sourse is 10.4.4.125...

**Comment by ghost** [comment]
I know on linux you can run this command to set ufw to allow all LAN connections:
sudo ufw allow out on eth1 to 10.0.0.0/24
If this is not possible on AFWall+ yet, can you implement a feature for this?

**Comment by ghost** [comment]
Hello..can any dev please reply to this issue?

**Comment by ildar** [comment]
I'm not a developer but willing to help.
I don't completely understand what works and what's not, sorry.
1. Do you confuse WiFi with LAN?
2. Are you sure it's the filtering problem, not routing?
Also could you please write a few examples of connections or pings working
and nonworking.
Also remember that the app can write logs which may be helpful. If
possible, please share

**Comment by ghost** [comment]
@ildar 
1. No, im talking about when a user is connected via his personal wifi, then its **his** LAN that im talking about.
2. I think you can call this a needed enhancement, but not an issue. This would be a feature added onto AFwall.

Under the following circumstance:
If user is connected to VPN via his personal WiFi and forces internet to connect only when VPN connected (which is currently the case), User would like to access his LAN **while** being connect to the VPN. 

Currently its impossible to access LAN unless user disconnects from VPN or a feature is implemented:
An option to allow/route all LAN traffic through the adapter (even though connected to VPN)

**Comment by walrus543** [comment]
I can confirm this issue. Another issue when using VPN is that I can't send MMS.
The dev is aware of these issues. Hopefully he will have a look soon.

**Comment by ghost** [comment]
yes @Primokorn i agree, i also cannot send MMS unless i disable AFWall & VPN all together.
